### PR TITLE
11.x improve resend transport response handling - fix

### PR DESCRIPTION
**Pull request by markovic-nikola**
Additional fix for the already closed PR: https://github.com/laravel/framework/pull/54004
After @jayanratna pointed out that successful API call doesn't have status code included.

Sorry for the extra work.

### CODE REVIEW

**Comment by jayanratna** [comment]
Just had a look at the Resend PHP SDK and it looks like I need to update the error list to catch the daily quota error. This is probably why you're having to add this logic in first place.

**Comment by jayanratna** [comment]
I have created this PR (https://github.com/resend/resend-php/pull/64) to update the Resend error list. This should allow the `daily_quota_exceeded` error to be correctly handled.

Currently getting it reviewed by the Resend team.

**Comment by markovic-nikola** [comment]
Thanks for the help!
I'll leave it to the Laravel team to decide whether to keep this PR as a 'fallback' solution or to revert it altogether.

File: src/Illuminate/Mail/Transport/ResendTransport.php [hunk 1]
```diff
@@ -99,7 +99,7 @@ protected function doSend(SentMessage $message): void
                 'attachments' => $attachments,
             ]);
 
-            throw_if($result['statusCode'] != Response::HTTP_OK, Exception::class, $result['message']);
+            throw_if(isset($result['statusCode']) && $result['statusCode'] != Response::HTTP_OK, Exception::class, $result['message']);
```
Comment (rodrigopedra) [inline src/Illuminate/Mail/Transport/ResendTransport.php:102]: I guess we should wrap the not equal comparison into parenthesis, as `&&` has a lower precedence than `!=`

```php
isset($result['statusCode']) && ($result['statusCode'] != Response::HTTP_OK)
```

reference: https://www.php.net/manual/en/language.operators.precedence.php

**Pull request merged**